pkg/api/v1alpha1/cluster_types.go

```diff
@@ -323,9 +325,48 @@ func (n *ClusterNetwork) Equal(o *ClusterNetwork) bool {
 	if n == nil || o == nil {
 		return false
 	}
+
+	if n.CNI != "" {
```

Comment:
shouldn't be enough to only check `CNIConfig`? at this point the cluster should already have the defaults assigned

Reply:
yeah that's what I was thinking, but when I was testing upgrade from an existing cluster, the eks-a controller at that point didn't have cniConfig set by the cluster defaults so it was in an error loop. I get that it will eventually progress once cluster defaults are set, but I thought it'll be better to have these additional checks

Reply:
Oh, is this because we don't have a mutation webhook, isn't it?

Reply:
yeah I think so

Reply:
cool, makes sense
what do you think about adding a comment here explaining exactly that?